morphuntion/resources/org/unicode/morphuntion/dictionary/dictionary_ar.lst

```diff
@@ -0,0 +1,63 @@
+مطعم: singular genitive nominative indefinite construct masculine participle active passive adjective noun verb inflection=7
```

Comment:
Similar to contractions comment - we should have a better explanation of fields. Do they need to be in some specific order? What are the possible values? Point to tools used to make them?

Reply:
The dictionary-parser historically generates these files, but in this contribution, it's only helpful as a reference. It will likely need to be replaced by a tool that generates equivalent data from Wikidata.

Everything before the colon is the word/phrase.  Everything after the colon and without an = is a white space separated list of grammemes. Anything after the colon is a list of key/value pairs. The key is not unique.  Think of it as a multi-value map.  Typically it's the key "inflection" with a value of the inflection table name.  The dictionary-parser typically enumerates the name. You can hand craft the name, but it's a pain to maintain such names.

Reply:
Related to this topic, any tool that generates this file will likely need a good way to filter the data.  Certain surface forms may be possible to automatically generate the data, like the English possessive/genitive form. **Most** abbreviations are unhelpful.  So you can omit them from the data.  Phrases are generally unhelpful, like "to the moon" in English.  Words that are technically correct, but are not widely understood outside the scope of a joke, like the singular form of sheep being shoop.

The order of the words don't matter, but it's easier to search the list when it's sorted.  When there are duplicate words, the second form overrides the first set of grammemes, except the key/value pairs.  For the key/value pairs, it prefixes them to the start of the set.  Typically the most popular key/value pair should be first.  So if there are 2 ways to interpret a word, like the noun "brush", the first one wins. When there is the same word, but the inflection tables have different parts of speech, usually the noun wins, but that can be overridden.  For the words "work" and "works", this distinction is important. See [morphuntion/test/resources/morphuntion/dialog/inflection/en.xml](https://github.com/unicode-org/inflection/pull/35/files#diff-47d33bf0724ecc058769400b4c5f702e9e058b9cbf7259db8b30d45871b4cbf7) lines 156-166 for an example of the caller overriding the inflection behavior for the given context.